src/rpc/rawtransaction.cpp

```diff
@@ -1791,6 +1792,320 @@ UniValue converttopsbt(const JSONRPCRequest& request)
+
+        for (const CTxIn& txin : psbtx.tx->vin) {
+            view.AccessCoin(txin.prevout); // Load entries from viewChain into view; can fail.
+        }
+
+        view.SetBackend(viewDummy); // switch back to avoid locking mempool for too long
+    }
+
+    // Fill the inputs
+    for (unsigned int i = 0; i < psbtx.tx->vin.size(); ++i) {
+        PSBTInput& input = psbtx.inputs.at(i);
+
+        if (input.non_witness_utxo || !input.witness_utxo.IsNull()) {
+            continue;
+        }
+
+        const Coin& coin = view.AccessCoin(psbtx.tx->vin[i].prevout);
+
+        std::vector<std::vector<unsigned char>> solutions_data;
+        txnouttype which_type = Solver(coin.out.scriptPubKey, solutions_data);
+        if (which_type == TX_WITNESS_V0_SCRIPTHASH || which_type == TX_WITNESS_V0_KEYHASH || which_type == TX_WITNESS_UNKNOWN) {
+            input.witness_utxo = coin.out;
+        }
+    }
+
+    CDataStream ssTx(SER_NETWORK, PROTOCOL_VERSION);
+    ssTx << psbtx;
+    return EncodeBase64((unsigned char*)ssTx.data(), ssTx.size());
+}
+
+UniValue joinpsbts(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() != 1)
+        throw std::runtime_error(
+                            "joinpsbts [\"psbt\",...]\n"
+                            "\nJoins multiple distinct PSBTs with different inputs and outputs into one PSBT with inputs and outputs from all of the PSBTs\n"
+                            "No input in any of the PSBTs can be in more than one of the PSBTs.\n"
+                            "\nArguments:\n"
+                            "1. \"txs\"                   (string) A json array of base64 strings of partially signed transactions\n"
+                            "    [\n"
+                            "      \"psbt\"             (string) A base64 string of a PSBT\n"
+                            "      ,...\n"
+                            "    ]\n"
+
+                            "\nResult:\n"
+                            "  \"psbt\"          (string) The base64-encoded partially signed transaction\n"
+                            "\nExamples:\n"
+                            + HelpExampleCli("joinpsbts", "\"psbt\"")
+                            );
+
+    RPCTypeCheck(request.params, {UniValue::VARR}, true);
+
+    // Unserialize the transactions
+    std::vector<PartiallySignedTransaction> psbtxs;
+    UniValue txs = request.params[0].get_array();
+    for (unsigned int i = 0; i < txs.size(); ++i) {
+        PartiallySignedTransaction psbtx;
+        std::string error;
+        if (!DecodePSBT(psbtx, txs[i].get_str(), error)) {
+            throw JSONRPCError(RPC_DESERIALIZATION_ERROR, strprintf("TX decode failed %s", error));
+        }
+        psbtxs.push_back(psbtx);
+    }
+
+    PartiallySignedTransaction merged_psbt(psbtxs[0]); // Copy the first one
+
+    // Clear signatures from all inputs
+    for (auto& input : merged_psbt.inputs) {
+        input.partial_sigs.clear();
```

Comment:
So this is necessary because the first is a copy and the remaining psbt inputs are cleared because of `AddInput`. Maybe remove this "optimization" and merge all psbts to an empty psbt?

Reply:
It would be less code, and less complexity, which I would say is always better, absent a reason to write more code. Why write more code?

Reply:
(Also, this is copied from combinepsbt, where it is easiest to treat the first tx specially, because we need something to compare all the others to, to make sure the underlying transaction is the same. That restriction doesn't exist here, so the need for the extra step is gone.)